### DISCUSSION TITLE
Omitting filename in concealed links [Help]

### DISCUSSION THREAD

**Discussion by wakelin-g** [zk-org/zk-nvim#139] [Help]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

Hi all. The vim conceal feature is quite cryptic to me, but is anyone aware of how the filename could be concealed in a wikilink-style link?

With `conceallevel=2`, I get wikilinks that look like this:

![image](https://github.com/mickael-menu/zk-nvim/assets/86983768/8f89942f-c4c8-48cc-af0c-67143450d084)

This is very nice! However, I would like them to look like this instead (i.e., the actual filename is hidden):

![image](https://github.com/mickael-menu/zk-nvim/assets/86983768/efa46755-dcc7-4d88-9db0-80ebafd771c7)

I am using the same conceal options copied from the readme:

```
" markdownWikiLink is a new region
syn region markdownWikiLink matchgroup=markdownLinkDelimiter start="\[\[" end="\]\]" contains=markdownUrl keepend oneline concealends
" markdownLinkText is copied from runtime files with 'concealends' appended
syn region markdownLinkText matchgroup=markdownLinkTextDelimiter start="!\=\[\%(\%(\_[^][]\|\[\_[^][]*\]\)*]\%( \=[[(]\)\)\@=" end="\]\%( \=[[(]\)\@=" nextgroup=markdownLink,markdownId skipwhite contains=@markdownInline,markdownLineStart concealends
" markdownLink is copied from runtime files with 'conceal' appended
syn region markdownLink matchgroup=markdownLinkDelimiter start="(" end=")" contains=markdownUrl keepend contained conceal
```

I _feel_ like this used to work automatically with my old config, but somewhere along the line some settings got changed and now it doesn't. If anyone has any ideas how to change this (or if you know that this can't be changed without an actual pr) please let me know.

Best,

GW

**Comment by nianiam** [comment]
Hello, this is fairly easy to do. You just tell the concealer to match a bit more of your link as the starting point:

**Old:**

```
syn region markdownWikiLink matchgroup=markdownLinkDelimiter start="\[\[" end="\]\]" ...
```

**New:**

```
syn region markdownWikiLink matchgroup=markdownLinkDelimiter start="\[\[\w\+|" end="\]\]" ...
```

This will match 1 more or more letters/numbers ([a-zA-Z0-9_]) and the pipe symbol in your WikiLink. If you are using a different ID/filename system you might need to change what you're matching. Or you might even need to be more restrictive. But this works for the default IDs.

I would suggest learning Vim's regex since it's a bit different to normal regex: https://vimregex.com

Hope that helps!
